metagraph/src/annotation/binary_matrix/row_sparse_disk/row_sparse_disk.hpp

```diff
@@ -0,0 +1,108 @@
+#pragma once
+
+#include <iostream>
+#include <string>
+#include <vector>
+
+#include "annotation/binary_matrix/base/binary_matrix.hpp"
+#include "common/vectors/bit_vector_adaptive.hpp"
+
+namespace mtg {
+namespace annot {
+namespace binmat {
+
+
+class RowSparseDisk : public BinaryMatrix {
+  public:
+    RowSparseDisk() {}
+    RowSparseDisk(const std::function<void(const RowCallback &)> &call_rows,
+                  uint64_t num_columns,
+                  uint64_t num_rows,
+                  uint64_t num_relations);
```

Comment:
instead of having to do the `const_cast` later to set the buffer size, why not initialize the buffer size here?